dotnet/src/webdriver/EncodedFile.cs

```diff
@@ -19,41 +19,41 @@
 
 using System;
 
+#nullable enable
+
 namespace OpenQA.Selenium
 {
     /// <summary>
     /// Represents a file transmitted over the wire as a base64-encoded string.
     /// </summary>
     public abstract class EncodedFile
     {
-        private string base64Encoded = string.Empty;
-        private byte[] byteArray;
-
         /// <summary>
         /// Initializes a new instance of the <see cref="EncodedFile"/> class.
         /// </summary>
         /// <param name="base64EncodedFile">The file as a Base64-encoded string.</param>
+        /// <exception cref="ArgumentNullException">If <paramref name="base64EncodedFile"/> is <see langword="null"/>.</exception>
+        /// <exception cref="FormatException">
+        /// <para>The length of <paramref name="base64EncodedFile"/>, ignoring white-space characters, is not zero or a multiple of 4.</para>
+        /// <para>-or-</para>
+        /// <para>The format of <paramref name="base64EncodedFile"/> is invalid. <paramref name="base64EncodedFile"/> contains a non-base-64 character,
+        /// more than two padding characters, or a non-white space-character among the padding characters.</para>
+        /// </exception>
         protected EncodedFile(string base64EncodedFile)
         {
-            this.base64Encoded = base64EncodedFile;
-            this.byteArray = Convert.FromBase64String(this.base64Encoded);
+            this.AsBase64EncodedString = base64EncodedFile ?? throw new ArgumentNullException(nameof(base64EncodedFile));
+            this.AsByteArray = Convert.FromBase64String(this.AsBase64EncodedString);
```

Comment:
We can defer computation: when user gets `AsByteArray` property getter, we can compute and return. Safe and minor change, as you wish (ideally as separate PR).

Reply:
Ah no, in this PR please. `Convert.FromBase64String(this.AsBase64EncodedString);` relies on the state (this.AsBase64EncodedString) which might be not computed (just prevent issues in future).

Reply:
I fixed this now to use the local ctor parameter.

However, I'm noticing there's a flaw here, since the array is writeable. Someone could change the value of the array and break things.

I vote to `[Obsolete]` the `AsByteArray` property, and instead add a `ToByteArray()` method which makes a new array each time. Targetting .NET 8, we can also make an `AsSpan` member which exposes the array as a `ReadOnlySpan<byte>`.

What do we think of this? I shouldn't do it in this PR, but if you think this is worthy of pursuit, I can open an issue for it.

Reply:
> I fixed this now to use the local ctor parameter.

Accepted.

> I vote to [Obsolete] the AsByteArray property, and instead add a ToByteArray()

You are right, I will accept it, as separate PR.

> we can also make an AsSpan

Oh not now, it is one more big step to make selenium modern, let's finish with AOT and NRT :D